sycl/source/detail/kernel_program_cache.hpp

```diff
@@ -112,11 +112,11 @@ class KernelProgramCache {
   };
 
   struct ProgramBuildResult : public BuildResult<ur_program_handle_t> {
-    AdapterPtr MAdapter;
-    ProgramBuildResult(const AdapterPtr &AAdapter) : MAdapter(AAdapter) {
+    const adapter_impl &MAdapter;
+    ProgramBuildResult(const adapter_impl &AAdapter) : MAdapter(AAdapter) {
```

Comment:
```suggestion
    ProgramBuildResult(const adapter_impl &Adapter) : MAdapter(Adapter) {
```

The whole point of `adapter_impl` renaming was to free up `Adapter` identifier.